src/Symfony/Component/ExpressionLanguage/ExpressionLanguage.php

```diff
@@ -84,7 +84,11 @@ public function parse($expression, $names)
             return $expression;
         }
 
-        $key = $expression.'//'.implode('-', $names);
+        $keys = array_map(function($key, $value){
+            return '{'."$key:$value".'}';
+        }, array_keys($names), $names);
```

Comment:
In fact, ideas like ``json_encode()``, ``serialize()`` and 
``implode('-', array_keys($names)).'-'.implode('-', $names)`` came to my mind first. But, like I said, the original naming convention is: **tokens separated by hyphens**. And I really don't want to break it to cause potential compatibility issues. 

Both ``json_encode()`` and ``serialize()`` break the convention because hyphens are missing, while imploding the keys looks fine but keys and values are lined up, in stead of paired, which is unnatural.
Besides, ``json_encode``  also produces different format for "indexed array" and "associative array", which introduces potential complexities. 
And as a proof of this side-effect,  if the test case is not altered, the PHPUnit test reports the following errors for ``json_encode``:
```
There was 1 failure:

1) Symfony\Component\ExpressionLanguage\Tests\ExpressionLanguageTest::testCachedParse
Expectation failed for method name is equal to <string:fetch> when invoked 2 time(s)
Parameter 0 for invocation Symfony\Component\ExpressionLanguage\ParserCache\ParserCacheInterface::fetch('1 + 1//[]') does not match expected value.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'1 + 1//'
+'1 + 1//[]'
```
So I finally decided to extend the naming convention by including {key:value} pairs in the generated cache key.